src/document/mod.rs

```diff
@@ -327,6 +144,30 @@ impl Document {
             index += 1;
         }
     }
+
+    /// Finds all file-specific entry points
+    pub fn get_entry_points<'a>(
```

Comment:
`get_` is a bit uncustomary for Rust. I think just `entry_points` would be fine.